api/utils/retry_promise.ts

```diff
@@ -0,0 +1,14 @@
+type RetryReturnType = (func: () => Promise<any>, retries: number, error?: Error) => Promise<any>;
+
+export const retryPromise: RetryReturnType = async (func, retries, error) => {
```

Comment:
```suggestion
export async function retryPromise (func: () => Promise<any>, retries: number, error?: Error) => {
```

Reply:
Had to do a manual type, but I implemented the spirit of your suggestion:

```
export async function retryPromise(
  func: () => Promise<any>,
  retries: number,
  error?: Error
): Promise<any> {
```